src/compiler/checker.ts

```diff
@@ -29665,9 +29665,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const elementFlags: ElementFlags[] = [];
         pushCachedContextualType(node);
         const inDestructuringPattern = isAssignmentTarget(node);
-        const inConstContext = isConstContext(node);
+        const isSpreadIntoCallOrNew = isSpreadElement(node.parent) && isCallOrNewExpression(node.parent.parent);
+        const inConstContext = isSpreadIntoCallOrNew || isConstContext(node);

```

Comment:
initially, I wanted to adjust `isConstContext` - but spread arguments never have contextual types + asking for a contextual type of an argument calls `getEffectiveCallArguments` and we are within this call already, so it was creating an infinite loop

I was looking into other ways to check for the const context for such spread expressions but I couldn't figure out anything since I can't even check the signature (it's still `resolvingSignature` at this point).